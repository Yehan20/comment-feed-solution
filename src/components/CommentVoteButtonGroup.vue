<script setup lang="ts"> 
import {  BiArrowUpCircle, BiArrowDownCircle } from "oh-vue-icons/icons";
import { addIcons } from 'oh-vue-icons';

addIcons(BiArrowDownCircle, BiArrowUpCircle);

  const props = defineProps<{
    upvoted: boolean,
    downvoted: boolean,

  }>()


  const emit = defineEmits<{
      '@upVote':[],
      '@downVote':[]
  }
  >()
</script>

<template>
    <button class="btn btn__upvote" :class="{ selected: props.upvoted }"
        :title="props.upvoted ? 'Click to Undo' : 'Click to Upvote'"

        @click="emit('@upVote')">

        <v-icon scale=1.2 name="bi-arrow-up-circle"></v-icon>
    </button>

    <button class="btn btn__downvote" :class="{ selected: props.downvoted }"
        @click="emit('@downVote')"
        :title="props.downvoted ? 'Click to Undo' : 'Click to DownVote'">

        <v-icon scale=1.2 name="bi-arrow-down-circle"></v-icon>
    </button>
</template>
<style lang="scss" scoped>
@use '../assets/scss/variables.scss' as v;

  button {
    text-transform: lowercase;
    border-radius: 20px;
    padding-left: 10px;
    padding-right: 10px;

    &.btn__upvote {

       padding: 5px;
       margin-right: 0;
   }

   &.btn__downvote {
       padding: 5px;
       margin-right: 0;

   }


   &.btn__upvote.selected {
       color: v.$brand-color !important;
   }

   &.btn__downvote.selected {
       color: v.$downvote-clr !important;
   }
  }

</style>