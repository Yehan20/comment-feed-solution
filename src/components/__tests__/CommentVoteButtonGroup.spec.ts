import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import CommentVoteButtonGroup from "../CommentVoteButtonGroup.vue";
import { addIcons, OhVueIcon, } from "oh-vue-icons";
import { BiArrowDownCircle, BiArrowUpCircle } from "oh-vue-icons/icons";


describe('CommentVoteButtonGroup', () => {


    beforeEach(() => {
        setActivePinia(createPinia());


    });

    // Mounting Test
    it("Component Should Render Correctly", () => {

        const wrapper = mount(CommentVoteButtonGroup, {

            props: {
                upvoted: true,
                downvoted: false,
            }, global: {
                components: {

                    'v-icon': OhVueIcon,  // Register v-icon manually in test

                }
            }
        })
        addIcons(BiArrowDownCircle, BiArrowUpCircle);

        expect(wrapper.html()).toMatchSnapshot()
    })

})
