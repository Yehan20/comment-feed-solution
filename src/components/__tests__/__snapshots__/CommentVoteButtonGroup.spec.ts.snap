// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentVoteButtonGroup > Component Should Render Correctly 1`] = `
"<button data-v-1090a549="" class="btn btn__upvote selected" title="Click to Undo"><svg data-v-1090a549="" class="ov-icon" style="font-size: 1.44em;" aria-hidden="true" width="23.04" height="23.04" viewBox="-1.6 -1.6 19.2 19.2" fill="currentColor">
    <path fill-rule="evenodd" d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zm-7.5 3.5a.5.5 0 01-1 0V5.707L5.354 7.854a.5.5 0 11-.708-.708l3-3a.5.5 0 01.708 0l3 3a.5.5 0 01-.708.708L8.5 5.707V11.5z"></path>
  </svg></button>
<button data-v-1090a549="" class="btn btn__downvote" title="Click to DownVote"><svg data-v-1090a549="" class="ov-icon" style="font-size: 1.44em;" aria-hidden="true" width="23.04" height="23.04" viewBox="-1.6 -1.6 19.2 19.2" fill="currentColor">
    <path fill-rule="evenodd" d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM8.5 4.5a.5.5 0 00-1 0v5.793L5.354 8.146a.5.5 0 10-.708.708l3 3a.5.5 0 00.708 0l3-3a.5.5 0 00-.708-.708L8.5 10.293V4.5z"></path>
  </svg></button>"
`;
